barretenberg/cpp/src/barretenberg/solidity_helpers/key_gen.cpp

```diff
@@ -69,18 +70,21 @@ int main(int argc, char** argv)
     if (plonk_flavour != "ultra") {
         info("Only ultra plonk flavour is supported at the moment");
         return 1;
+    }
+
+    info("Generating ultra plonk keys for ", circuit_flavour, " circuit");
+
+    if (circuit_flavour == "blake") {
+        generate_keys<UltraComposer, BlakeCircuit>(output_path, plonk_flavour, circuit_flavour);
+    } else if (circuit_flavour == "add2") {
+        generate_keys<UltraComposer, Add2Circuit>(output_path, plonk_flavour, circuit_flavour);
+    } else if (circuit_flavour == "recursive") {
+        generate_keys<UltraComposer, RecursiveCircuit>(output_path, plonk_flavour, circuit_flavour);
+    } else if (circuit_flavour == "ecdsa") {
+        generate_keys<UltraComposer, EcdsaCircuit>(output_path, plonk_flavour, circuit_flavour);
     } else {
-        info("Generating ultra plonk keys for ", circuit_flavour, " circuit");
-
-        if (circuit_flavour == "blake") {
-            generate_keys<UltraComposer, BlakeCircuit>(output_path, plonk_flavour, circuit_flavour);
-        } else if (circuit_flavour == "add2") {
-            generate_keys<UltraComposer, Add2Circuit>(output_path, plonk_flavour, circuit_flavour);
-        } else if (circuit_flavour == "recursive") {
-            generate_keys<UltraComposer, RecursiveCircuit>(output_path, plonk_flavour, circuit_flavour);
-        } else {
-            info("Only blake, add2 and recursive circuits are supported at the moment");
-            return 1;
-        }
+        info("Only blake, add2 and recursive circuits are supported at the moment");
```

Comment:
We should probably change this to just be "unsupported circuit" so we don't have to keep extending it with specific names.